### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by ha-manel**
## In this PR I:
- [x] Created model classes for all entities.
- [x] Set up associations between models.
- [x]  Created automatic rails counters (posts_counter, likes_counter, comments_counter).
- [x] Inserted data into tables.
- [x] Created custom methods in the models.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (FrederickMih) [inline app/models/user.rb:4]: - [Optional] You may also be considered adding foreign_keys, that's

```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

**Review by FrederickMih — APPROVED** [review]
Hi @ha-manel , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (ha-manel) [reply app/models/user.rb:4]: Hello @FrederickMih, thank you for the review. It's already added in `foreign_key: 'author_id'` no? 😅

Comment (ha-manel) [reply app/models/user.rb:4]: Hello @FrederickMih, thank you for the review! I think the user foreign keys should be added in the `posts`, `comments`, and `likes` models not in `users`. Please correct me if I'm wrong.

**Pull request merged**